src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/RegularExpressionAttribute.cs

```diff
@@ -31,6 +31,29 @@ public RegularExpressionAttribute([StringSyntax(StringSyntaxAttribute.Regex)] st
         /// </summary>
         public int MatchTimeoutInMilliseconds { get; set; }
 
+        /// <summary>
+        /// Gets or sets the timeout to use when matching the regular expression pattern
+        /// </summary>
+        public TimeSpan MatchTimeout
+        {
+            get => TimeSpan.FromMilliseconds(MatchTimeoutInMilliseconds);
+            set
+            {
+                long timeoutMilliseconds = (long)value.TotalMilliseconds;
```

Comment:
I suggest to just do what is good enough for Task today:
https://github.com/dotnet/runtime/blob/7db092a6ed68a5fbf47962c45bb0c1bb965d02c6/src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs#L2642-L2646

that will be half the strings, and less duplicated code.

if it's important to tell them what's wrong, you could include the value in the string. (Task doesn't bother)

Reply:
I removed the setter because we don't need it

Reply:
right, but you're still repeating this validation code. my suggestion is to replace it everywhere with just the 5 lines above, effectively.

Reply:
then you don't need new strings (or just 1 if you choose to include the value, which is nice)